src/main/java/io/cdap/e2e/utils/GcpClient.java

```diff
@@ -63,6 +70,16 @@ public static int executeQuery(String query) throws InterruptedException, IOExce
     return 0;
   }
 
+  public static Optional<String> executeQueryOnBQ(String query) throws InterruptedException, IOException {
+    QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(query).build();
+    TableResult results = getBigQueryService().query(queryConfig);
+    String outputRowValue = null;
+    if (results.getTotalRows() > 0) {
+      outputRowValue =  (String) results.getValues().iterator().next().get(0).getValue();
```

Comment:
One more question, as far as I can see this will return the first value of the first row of the table if I am not wrong? Why do we need only that value? It earlier made sense because we were using this method to execute count and drop query. So, I think it would be better to return the whole resultSet and the caller can fetch the required data from the result set! What do you think @tivv

Reply:
as far as I understand the whole goal is to get first item as this is the use case. May be that's also an answer to the method name, e.g. `getSoleQueryResult`

Reply:
Changed method name to `getSoleQueryResult`